python/tk_desktop/install_apps.py

```diff
@@ -0,0 +1,250 @@
+    "button" commands are found for the current project. Displays site
+    Software entity icons and a link to a Shotgun Support article detailing
+    how to configure Software entities.
+    """
+    def __init__(self, parent=None):
+        QtGui.QWidget.__init__(self, parent)
+
+        self.ui = install_apps.Ui_InstallApps()
+        self.ui.setupUi(self)
+
+        # Assign magic numbers and such
+        self._icon_size = QtCore.QSize(64, 64)
+        self._icon_layout_spacing = 5
+        self._icon_layout_stretch = 1
+        self._icon_column_count = 4
+        self._config_link_color = "#0000FF"
+        self._config_link_text = "here"
+
+        # Hide this widget by default.
+        self.setVisible(False)
+
+    def build(self, project=None):
+        """
+        Query Shotgun for a list of Software images to display and
+        set the Software configuration link value
+
+        :param project: Current Shotgun Project instance (dict)
+        """
+        # Make sure we can get a handle to the current engine.
+        # This is primarily used for logging purposes.
+        engine = sgtk.platform.current_engine()
+        if not engine:
+            return
+
+        self._load_software_icons(engine, project)
+        self._set_configuration_link(engine)
+
+    def _clear_icons(self):
+        """
+        Remove exising icons to prepare for new icons.
+        This is necessary because the list of icons may not
+        be the same from project to project, or can change
+        if Software entities are reconfigured in Shotgun.
+        """
+        while self.ui.icon_rows.count():
+            qt_layout = self.ui.icon_rows.takeAt(0)
+            while qt_layout.count():
+                qt_item = qt_layout.takeAt(0)
+                qt_widget = qt_item.widget()
+                if qt_widget is not None:
+                    qt_widget.deleteLater()
+            qt_layout.deleteLater()
+
+    def _load_software_icons(self, engine, project=None):
+        """
+        Displays thumbnails downloaded from Software entities for a
+        specified Project in the ui.icon_rows layout.
+
+        :param engine: Toolkit engine instance.
+        :param project: Shotgun Project instance or None
+        """
+        # Get the list of Software entities for this Project from Shotgun
+        sg_softwares = self._get_sg_software_entities(engine, project)
+        if not sg_softwares:
+            return
+
+        # Clear out any existing icons
+        self._clear_icons()
+
+        # Arrange the icons in a grid with self._icon_column_count
+        # columns and N rows based on the total number of icons
+        col_num = 0
+        icon_row_layout = None
+        last_software = len(sg_softwares) - 1
+        for i in range(len(sg_softwares)):
+            # Download the thumbnail from Shotgun
+            sg_icon = shotgun_data.ShotgunDataRetriever.download_thumbnail(
```

Comment:
Fixing the transparency is important, but I think the loading thumbnail can wait. We can polish this once we are sure everything else that's important has made it.

With that being said... visual thumbnail states (not loading, loading, loaded, error) are something I would LOOOOOOVE to see happen.